source/driver_run.cpp

```diff
@@ -29,18 +29,23 @@ void Driver::driver_run(void)
     ModuleESolver::init_esolver(p_esolver);
 
     //! 2: setup cell and atom information
+
+    // this warning should not be here, mohan 2024-05-22
 #ifndef __LCAO
     if(GlobalV::BASIS_TYPE == "lcao_in_pw" || GlobalV::BASIS_TYPE == "lcao")
     {
         ModuleBase::WARNING_QUIT("driver","to use LCAO basis, compile with __LCAO");
     }
 #endif
+
+    // the life of ucell should begin here, mohan 2024-05-12
+    // delete ucell as a GlobalC in near future
     GlobalC::ucell.setup_cell(GlobalV::stru_file, GlobalV::ofs_running);
 
     //! 3: initialize Esolver and fill json-structure 
-    p_esolver->init(INPUT, GlobalC::ucell);
-
+    p_esolver->before_all_runners(INPUT, GlobalC::ucell);
 
+    // this Json part should be moved to before_all_runners, mohan 2024-05-12
```

Comment:
Maybe it is not the function of esolver, but the function of ucell. So it should not be moved to before_all_runners.

Reply:
I agree this part can be further discussed.